sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -352,7 +373,29 @@ static size_t _handle_req(coap_pkt_t *pdu, uint8_t *buf, size_t len,
         return -1;
     }
 
-    ssize_t pdu_len = resource->handler(pdu, buf, len, resource->context);
+    /* perform a cache lookup if compiled with module nanocoap_cache */
```

Comment:
@kb2ma this change to `gcoap` is highly dependent on how we proceed with #13790. I will comment there shortly. But maybe you already see a nicer integration than the presented method?